lisa/microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -1338,6 +1426,707 @@ def _set_block_size_env_var(self, testcase: str) -> None:
+            match = re.search(r"dev\s+(\S+)", result.stdout)
+            if not match:
+                self._log.debug(
+                    f"Could not parse primary NIC from: {result.stdout.strip()}"
+                )
+                return
+
+            primary_nic = match.group(1)
+
+            # Get current MTU
+            current_mtu_result = self.node.execute(
+                f"ip link show {primary_nic}",
+                shell=True,
+            )
+
+            if current_mtu_result.exit_code == 0:
+                # Extract MTU using Python regex
+                # Expected format: "... mtu 1500 ..."
+                mtu_match = re.search(r"mtu\s+(\d+)", current_mtu_result.stdout)
+                if not mtu_match:
+                    self._log.debug(
+                        f"Could not parse MTU from: {current_mtu_result.stdout.strip()}"
+                    )
+                    return
+
+                current_mtu = int(mtu_match.group(1))
+
+                if current_mtu != expected_mtu:
+                    # Set MTU
+                    self.node.execute(
+                        f"sudo ip link set {primary_nic} mtu {expected_mtu}",
+                        shell=True,
+                        sudo=True,
+                    )
+
+    def _log_thermal_health(self, stage: str) -> None:
+        """
+        Log thermal health: sensors, throttle counts, frequency state.
+
+        Args:
+            stage: Execution stage (preflight, anchor_failure, suite_end)
+        """
+        self._log.info(f"=== Thermal Health [{stage}] ===")
+
+        # sensors output (if available)
+        result = self.node.execute(
+            "sensors | head -20",
+            shell=True,
+        )
+        if result.exit_code == 0 and result.stdout.strip():
+            for line in result.stdout.split("\n")[:10]:
+                if line.strip():
+                    self._log.info(f"  {line}")
+        else:
+            self._log.info("  Sensors: not available")
+
+        # Throttle counts
+        result = self.node.execute(
+            "for f in /sys/devices/system/cpu/cpu*/thermal_throttle/"
+            "*_throttle_count; do "
+            '[ -f "$f" ] && echo "$f: $(cat $f)"; done | head -5',
+            shell=True,
+        )
+        if result.exit_code == 0 and result.stdout.strip():
+            self._log.info(f"  Throttle counts: {result.stdout.strip()}")
+        else:
+            self._log.info("  Throttle counts: not available")
+
+        # CPU frequency (sample first 4 CPUs)
+        result = self.node.execute(
+            "for cpu in 0 1 2 3; do "
+            "f=/sys/devices/system/cpu/cpu$cpu/cpufreq/scaling_cur_freq; "
+            '[ -f "$f" ] && echo "CPU$cpu: $(cat $f) kHz"; done',
+            shell=True,
+        )
+        if result.exit_code == 0 and result.stdout.strip():
+            self._log.info(f"  Frequencies: {result.stdout.strip()}")
+
+    def _run_warmup_for_test(self, testcase: str) -> None:
```

Comment:
The CH perf tests framework already runs each test for multiple iterations and calculates the mean, standard deviation etc. This should already cancel out any effects of the system not being warmed up etc. In the worst-case scenario, it should be sufficiently warmed up after the first iteration.

Do you have comprehensive data show that doing this warmup helps?